test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/TempDataApplicationModelProviderTest.cs

```diff
@@ -110,13 +95,9 @@ public void OnProvidersExecuting_SetsKeyPrefixToEmptyString_IfCompatSwitchIsSet(
             // Arrange
             var expected = typeof(TestController_OneTempDataProperty).GetProperty(nameof(TestController_OneTempDataProperty.Test2));
             var options = Options.Create(new MvcViewOptions { SuppressTempDataAttributePrefix = true });
+            var type = typeof(TestController_OneTempDataProperty);
             var provider = new TempDataApplicationModelProvider(options);
-            var defaultProvider = new DefaultApplicationModelProvider(
-                Options.Create(new MvcOptions()),
-                TestModelMetadataProvider.CreateDefaultProvider());
-
-            var context = new ApplicationModelProviderContext(new[] { typeof(TestController_OneTempDataProperty).GetTypeInfo() });
-            defaultProvider.OnProvidersExecuting(context);
+            var context = GetContext(type);
```

Comment:
Pass in options from line no. 97 to this call and in other places in this PR. 

FYI: We have a general clean up item for model binding: https://github.com/aspnet/Mvc/issues/7317

Reply:
Oops. Ignore it. Read it too fast. line no. 97 is for MvcViewOptions and not MvcOptions